src/Microsoft.XmlSerializer.Generator/src/Microsoft/XmlSerializer/Generator/XmlSerializationReader.cs

```diff
@@ -2217,7 +2220,8 @@ private void WriteMemberElementsIf(Member[] members, Member anyElement, string e
                     ElementAccessor e = elements[j];
                     string ns = e.Form == XmlSchemaForm.Qualified ? e.Namespace : "";
                     if (!isSequence && e.Any && (e.Name == null || e.Name.Length == 0)) continue;
-                    if (!firstElement || (!isSequence && count > 0))
+
+                    if(!firstElement && isSequence)
```

Comment:
I don't think there's any case we want to write `else`. And when would this condition be `true`?

Reply:
All the changes are only for the case when isSequence = false.